worlds/stardew_valley/logic.py

```diff
@@ -1,500 +1,534 @@
+            FestivalCheck.legend_of_the_winter_star: self.has_season(Season.winter) & self.can_reach_region(Region.town),
         })
 
+        self.special_order_rules.update({
+            SpecialOrder.island_ingredients: self.has_island_transport() & self.can_farm_perfectly() &
+                                  self.has(Vegetable.taro_root) & self.has(Fruit.pineapple) & self.has(Forageable.ginger),
+            SpecialOrder.cave_patrol: self.can_mine_perfectly() & self.can_mine_to_floor(120),
+            SpecialOrder.aquatic_overpopulation: self.can_fish_perfectly(),
+            SpecialOrder.biome_balance: self.can_fish_perfectly(),
+            SpecialOrder.rock_rejuivenation: self.has(Mineral.ruby) & self.has(Mineral.topaz) & self.has(Mineral.emerald) &
+                                 self.has(Mineral.jade) & self.has(Mineral.amethyst) & self.has_relationship(NPC.emily, 4) &
+                                 self.has(ArtisanGood.cloth) & self.can_reach_region(Region.haley_house),
+            SpecialOrder.gifts_for_george: self.has_season(Season.spring) & self.has(Forageable.leek),
+            SpecialOrder.fragments_of_the_past: self.can_reach_region(Region.dig_site),
+            SpecialOrder.gus_famous_omelet: self.has(AnimalProduct.any_egg),
+            SpecialOrder.crop_order: self.can_farm_perfectly(),
+            SpecialOrder.community_cleanup: self.can_crab_pot(),
+            SpecialOrder.the_strong_stuff: self.can_keg(Vegetable.potato),
+            SpecialOrder.pierres_prime_produce: self.can_farm_perfectly(),
+            SpecialOrder.robins_project: self.can_chop_perfectly() & self.has(Material.hardwood),
+            SpecialOrder.robins_resource_rush: self.can_chop_perfectly() & self.has(Fertilizer.tree) & self.can_mine_perfectly(),
+            SpecialOrder.juicy_bugs_wanted_yum: self.has(Loot.bug_meat),
+            SpecialOrder.tropical_fish: self.has_island_transport() & self.has(Fish.stingray) & self.has(Fish.blue_discus) & self.has(Fish.lionfish),
+            SpecialOrder.a_curious_substance: self.can_mine_perfectly() & self.can_mine_to_floor(80),
+            SpecialOrder.prismatic_jelly: self.can_mine_perfectly() & self.can_mine_to_floor(40),
+            SpecialOrder.qis_crop: self.can_farm_perfectly() & self.can_reach_region(Region.greenhouse) &
+                         self.can_reach_region(Region.island_west) & self.has_total_skill_level(50) &
+                         self.has(Machine.seed_maker),
+            SpecialOrder.lets_play_a_game: self.has_junimo_kart_max_level(),
+            SpecialOrder.four_precious_stones: self.has_lived_months(MAX_MONTHS) & self.has("Prismatic Shard") &
+                                    self.can_mine_perfectly_in_the_skull_cavern(),
+            SpecialOrder.qis_hungry_challenge: self.can_mine_perfectly_in_the_skull_cavern() & self.has_max_buffs(),
+            SpecialOrder.qis_cuisine: self.can_cook() & self.can_spend_money(250000),
```

Comment:
I believe this needs some kind of access requirement either to Pierre's or the Saloon